service/worker/scheduler/calendar_test.go

```diff
@@ -91,36 +104,97 @@ func (s *calendarSuite) TestCalendarMatch() {
 	s.True(cc.matches(time.Date(2022, time.March, 16, 12, 55, 55, 0, time.UTC)))
 }
 
+func (s *calendarSuite) TestParseCronString() {
+	scs, iv, tz, err := parseCronString(&schedpb.CronString{CronSpec: "5,9 */2 * * *"})
+	s.NoError(err)
+	s.Equal(&schedpb.StructuredCalendarSpec{
+		Second:     []*schedpb.Range{{Start: 0}},
+		Minute:     []*schedpb.Range{{Start: 5}, {Start: 9}},
+		Hour:       []*schedpb.Range{{Start: 0, End: 23, Step: 2}},
+		DayOfMonth: []*schedpb.Range{{Start: 1, End: 31}},
+		Month:      []*schedpb.Range{{Start: 1, End: 12}},
+		DayOfWeek:  []*schedpb.Range{{Start: 0, End: 7}},
```

Comment:
DayOfMonth starts from 1, DayOfWeek starts from 0?

Reply:
And Should it end with 6 here? I know both 0 and 7 are for Sunday, but still.

Reply:
yes.

and yeah, it should, kinda. it was easier to just ignore the special case and leave it as 7, which is harmless. but I think I can fix it without too much work.

Reply:
Or maybe Start: 1, End: 7, and 0 is just a special value for Sunday.